### PR TITLE
Added CMake + updated github actions workflow

### PR DESCRIPTION
**Pull request by TheTryton**
Added CMake for gi_wrapper_cpp subproject (GDCM library)

Updated github actions workflow to include GDCM in dependency list in vcpkg and linux package manager

### CODE REVIEW

**Comment by TheTryton** [comment]
Do not merge currently as there are some semantical code errors to fix

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp]: @kdbogusz please be aware that _int8 is not a standard fixed width integer type (if I'm not wrong it is exclusive to MSVC), please use int8_t from <cstdint> as it is listed as standardized fixed width integer type (please refer to https://en.cppreference.com/w/cpp/types/integer)

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp]: @kdbogusz Same as above (change _int16 to int16_t)

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp]: @kdbogusz  Same as above (change _int32 to int32_t)

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp]: @kdbogusz Same as above (change _int64 to int64_t)

File: gi_wrapper_cpp/dicom_tag_strategies.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_tag_strategies.cpp:102]: @kdbogusz It would be nice to extract this 0x7FE0 constant to some named constant like:
constexpr int64_t <name-of-constant> = 0x7FE0;
This should apply to most of the constants + they all should be placed in another file (preferable header file)

File: gi_wrapper_cpp/wrapper.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/wrapper.cpp:45]: @kdbogusz that is a potential access to uninitialized memory - gdcm documentation doesn't specify ownership rules to the returned file thus I would assume that it is owned by the reader object thus after exiting loadDicom scope the referenced file object returned by this function is destroyed (as the reader is destroyed) - because of that I would suggest returning the file object by value (File loadDicom(...)) or copy that to heap and return as unique_ptr (less preferable)

File: gi_wrapper_cpp/dicom_tag_strategies.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_tag_strategies.cpp:92]: @kdbogusz please pass userTags by const reference - current signature causes unnecessary copy of potentially big data structure (const set<pair<int, int>>& userTags)

Comment (TheTryton) [reply gi_wrapper_cpp/dicom_tag_strategies.cpp:92]: This should apply to all places where such data structures are passed (including unchanged code)

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp:172]: @kdbogusz As said in other comment please extract those constants as some compile time constants (constexpr ...) to indicate the meaning of them

File: gi_wrapper_cpp/wrapper_logger.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/wrapper_logger.cpp:33]: @kdbogusz please change old linux-style timestamp taking to newer more portable way of doing that - see https://en.cppreference.com/w/cpp/chrono

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp:85]: @kdbogusz 1. Never use new explicitly 2. this array is never deallocated 3. for this purpose use either std::vector of requested size or unique_ptr<char[]>

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (TheTryton) [inline gi_wrapper_cpp/dicom_reader.cpp:84]: This explicitly allocated array exits function scope - this may be a memory leak if somebody forgets to deallocate that when using this function - please change the signature of the function to return std::vector<double> instead of some double pointer

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (grzanka) [inline gi_wrapper_cpp/dicom_reader.cpp]:
```suggestion
                    int8_t currentValue;
```

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (grzanka) [inline gi_wrapper_cpp/dicom_reader.cpp]:
```suggestion
                    int16_t currentValue;
```

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (grzanka) [inline gi_wrapper_cpp/dicom_reader.cpp]:
```suggestion
                    int32_t currentValue;
```

File: gi_wrapper_cpp/dicom_reader.cpp
Comment (grzanka) [inline gi_wrapper_cpp/dicom_reader.cpp]:
```suggestion
                    int64_t currentValue;
```

File: .github/workflows/test.yml
Comment (grzanka) [inline .github/workflows/test.yml]:
```suggestion
      run: |
           sudo apt update
           sudo apt install libwxgtk3.0-gtk3-dev qt5-default libgdcm3.0 libgdcm-dev
```

**Comment by kdbogusz** [comment]
Checks passed, however I have two disclaimers:

1. `/examples` causes some problems with build, because it uses make instead of cmake, which as far as I know we don't want and I don't know how to attach a library installed via apt-get to it. At the moment I commented out lines responsible for building it in `deploy.sh` and it probably needs a more serious refactor.
2. Only a handful of issues mentioned it the review have been fixed, the rest I can either fix before merging or in a different PR. If I remember correctly this merge is blocking something else, so I lean towards the second option.

**Review by kdbogusz — APPROVED** [review]
Approved by @TheTryton

**Pull request merged**
